bbq/vm/value_function.go

```diff
@@ -30,7 +30,8 @@ import (
 
 type FunctionValue interface {
 	interpreter.FunctionValue
-	GetParameterCount() int
+	HasDerivedType() bool
+	DerivedFunctionType(receiver Value, context interpreter.TypeConverter) *sema.FunctionType
```

Comment:
Maybe add comments here explaining what "derived" means. Documenting the cases with some examples, like discussed in the Impl. Sync, might help

Reply:
Good idea! Added some documentation: https://github.com/onflow/cadence/pull/3897/commits/29dbed0f9e45ee8a562ce934b9dbcd89083aa8f6

While at it, also renamed the function to `HasGenericType`, since it seems to be more intuitive.